example-component/setup.py

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env python
+# -- coding: utf-8 --
+# Copyright 2017 Reactive Ops Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import sys
+from setuptools import setup, find_packages
+
+# read in the variables defined in lib/release as global
+# to be used below
+execfile('release.py')
```

Comment:
What is the reasoning for using `execfile` here instead of an `import`? Are there cases where we need the settings file to be executed logic instead of a declarative format?

Reply:
execfile imports the variables into the current context. import forces it to be used in the module context. it `pentagoncomponent.__version__` instead of `__version__`.

I made this change a few months ago in the pentagon package because it made something I was doing better. I don't recall what that is at this time.

Reply:
I'd prefer it namespace into the lib to avoid any collisions that could cause problems. The `setup.py` file likely won't be modified much right?

Reply:
I know there was an important reason I did it at the time. I'll try again and see if I can sort out what it was. It might not be necessary anymore.

Reply:
Ah, broke it and found the reason.
When you do `from pentagon.release import __version__,__author__` it *also* imports `pentagon/init.py` which breaks because `init.py` imports modules that haven't been installed yet. 

`execfile()` got around that. Do you have suggestions regarding how to import `release` and allowing it to finish running `pip install`?